sensor/tests/resource/helper.go

```diff
@@ -378,17 +404,17 @@ func (c *TestContext) LastViolationStateByID(id string, assertion AlertAssertFun
 func (c *TestContext) LastViolationStateByIDWithTimeout(id string, assertion AlertAssertFunc, message string, checkEmptyAlertResults bool, timeout time.Duration) {
 	timer := time.NewTimer(timeout)
 	ticker := time.NewTicker(10 * time.Millisecond)
-	var lastErr error
+	lastErr := errors.New("no alerts asserted")
```

Comment:
IIUC error won't be set if there are no violations with _this deployment ID_. Therefore, I think the message can be a bit clearer: 
```suggestion
	lastErr := errors.Errorf("no alerts sent for deployment ID %s", id)
```

Reply:
Done